src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/controller/InterventionController.kt

```diff
@@ -0,0 +1,25 @@
+package uk.gov.justice.digital.hmpps.hmppsinterventionsservice.controller
+
+import org.springframework.http.HttpStatus
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RestController
+import org.springframework.web.server.ResponseStatusException
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.dto.InterventionDTO
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.service.InterventionService
+import uk.gov.justice.digital.hmpps.hmppsinterventionsservice.utils.StringToUUIDConverter
+
+@RestController
+class InterventionController(
+  private val interventionService: InterventionService,
+) {
+
+  @GetMapping("/intervention/{id}")
+  fun getInterventionByID(@PathVariable id: String): InterventionDTO {
+    val uuid = StringToUUIDConverter.parseID(id)
+
+    return interventionService.getIntervention(uuid)
+      ?.let { InterventionDTO.from(it) }
+      ?: throw ResponseStatusException(HttpStatus.NOT_FOUND, "intervention not found [id=$uuid]")
```

Comment:
We have the opportunity to drop the message if you'd like

- "intervention not found" is implicit from 404 Not Found on a call to /intervention/{id}
- the id is also implicit in the caller context

Reply:
true  - will update

Reply:
this follows existing patterns. the message is formatted into an error response that can be used by the UI. would prefer to keep it consistent.

Reply:
ok - will put back in